### ISSUE TITLE
Logout does not work

### ISSUE THREAD

**Issue by mcada** [syndesisio/syndesis#3529]
## This is a...

<pre><code>
[ ] Feature request
[x] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

Log out does nothing. It works on 1.4.6 and 1.4.7.

Syndesis version: 
Ignite:
    1.4.7.fuse-710001-redhat-00001

There is an error in console:

![screenshot_20180831_142627](https://user-images.githubusercontent.com/14313995/44912257-deeae300-ad29-11e8-91e3-f1d1d866ba72.png)

**Comment by heiko-braun** [comment]
@mcada You are saying it works on works on `1.4.6` and `1.4.7` (upstream versions), but fails on `1.4.7.fuse-710001-redhat-00001` (prod version) ?

**Comment by heiko-braun** [comment]
@syndesisio/ui-api Can anybody take a look and help diagnose this issue?

**Comment by heiko-braun** [comment]
@mcada Looks like you've tested on minishift. Can you verify on OCP as well? If it works on OCP we can lower the priority.

**Comment by gashcrumb** [comment]
What browser is this on?

**Comment by gashcrumb** [comment]
Also just to be on the safe side if you clear your cache is it still reproducible.  

The error here kinda makes me concerned that this is a bad console build almost, or was built after a bad merge omitted some code.

**Comment by heiko-braun** [comment]
We need to confirm:

- What's the environment this happens on?
- What do users see/experience?
- Does a work around exist?
- Does logging out of openshift console help?

**Comment by gashcrumb** [comment]
I guess the other question is how reproducible this is, I could see maybe in some circumstance [this line](https://github.com/syndesisio/syndesis/blob/1.4.x/app/ui/src/app/core/providers/user-provider.service.ts#L44) could trigger that error.

**Comment by mcada** [comment]
I used minishift and firefox for testing. On 1.4.7 I am redirected to logout endpoint correctly.
On 1.4.7.fuse-710001-redhat-00001 I am not redirected and syndesis page looks empty, clicking on any navigation button behaves as usual - everything works. Logging out from console did not log me out from syndesis. I will try OCP.

**Comment by gashcrumb** [comment]
Can you also provide a screenshot of the network tab when this occurs?

**Comment by mcada** [comment]
This is how it looks like when I click on logout:

![screenshot_20180831_154248](https://user-images.githubusercontent.com/14313995/44915890-8a993080-ad34-11e8-9cad-591feafc4f66.png)

**Comment by gashcrumb** [comment]
Also does this behave on chrome for you?  Can you try firefox on another machine out of curiosity?

**Comment by mcada** [comment]
The issue is present on both firefox and chrome on different machines and minishift.

**Comment by heiko-braun** [comment]
@mcada Do you know of any workaround?

**Comment by mcada** [comment]
Well, you can delete all cookies and it will stop working until you log in again.

**Comment by heiko-braun** [comment]
It's not, by chance, related to `Adblock Plus` enabled enabled?

**Comment by mcada** [comment]
No, it does not work with disabled adblock. And it works on 1.4.7 with enabled adblock

**Comment by gashcrumb** [comment]
So what's not shown is that in fact the browser does actually go to `/logout` and is redirected back, unfortunately the `logout.html` page is not retrieved.  So I would say that the problem here could be this config -> https://github.com/syndesisio/syndesis/pull/3261/files#diff-8c5f4d368563a6cebe49737dba5dd301R7

**Comment by gashcrumb** [comment]
So running the rebuilt image looks like:

![image](https://user-images.githubusercontent.com/351660/44927267-e4442f80-ad21-11e8-9839-a55c44498c0b.png)

Now clicking logout I see:

![image](https://user-images.githubusercontent.com/351660/44927284-0047d100-ad22-11e8-9ca9-7d9a70d27fa0.png)

In our nginx config for the UI image we have this setting, which (I'm no nginx guy) I'm assuming ensures that `logout.html` is returned when this endpoint is requested:

https://github.com/syndesisio/syndesis/pull/3261/files#diff-8c5f4d368563a6cebe49737dba5dd301L14

However at that path in the UI pod there's only:

![image](https://user-images.githubusercontent.com/351660/44927372-53ba1f00-ad22-11e8-98a6-097e3340159b.png)

The actual file resides here:

![image](https://user-images.githubusercontent.com/351660/44927393-6af90c80-ad22-11e8-89c1-99c07233e701.png)

So the question I don't know the answer to is, why's this different?  Do we need a product image specific nginx config?

**Comment by gashcrumb** [comment]
Sure enough checking in the upstream image by rolling back from the productized image version, the assets are at `/usr/share/nginx/html` as I'd expect, so the nginx config is then referencing the correct file and logout works properly.

**Comment by gashcrumb** [comment]
@heiko-braun so coming full circle, normally the static assets for the UI are copied to `/usr/share/nginx/html` [per the dockerfile for the UI](https://github.com/syndesisio/syndesis/blob/1.4.x/app/ui/docker/Dockerfile#L60), but the downstream version apparently needs to put these in a different location (`/opt/app-root`) for some reason that I'm unaware of.  Maybe a different base image?  

Seems like in any case we should align these locations across the two builds so as to avoid having to carry a patched nginx config in the product branch.

**Comment by mcada** [comment]
Redirected to logout endpoint now works correctly. However this fix introduced older issue which is only present on firefox:
![out](https://user-images.githubusercontent.com/14313995/44975578-1e504400-af63-11e8-853a-17f3d5c95243.gif)


I already reported it some time ago: https://github.com/syndesisio/syndesis/issues/3016
This is not a blocker though as refreshing the page redirects user to login page.

**Comment by aileenc** [comment]
Can this be marked as resolved or downgraded as no longer a blocker.
